cabal-install/src/Distribution/Client/Nix.hs

```diff
@@ -141,6 +141,10 @@ nixShell verb dist globalFlags config go = do
   if alreadyInShell
     then go
     else do
+      -- Nix integration never worked with cabal-install v2 commands ...
+      print "WARNING: Nix integration has been deprecated and will be removed in a future release."
+      print "You can learn more about it here: https://cabal.readthedocs.io/en/latest/nix-integration.html"
```

Comment:
You can actually use `warn verbosity "msg"` instead of print! :) This way it integrates better with the verbosity of cabal-install

Reply:
Almost, the `warn` function introduces a `Warning: ` prefix already, so you can just write your message